dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/user/DefaultUserService.java

```diff
@@ -1,714 +1,721 @@
-            .getUserAuthorityGroups().stream().map( BaseIdentifiableObject::getUid ).collect( Collectors.toList() ) );
-
-        roles.forEach( ur ->
-        {
-            if ( !currentUser.getUserCredentials().canIssueUserRole( ur, canGrantOwnUserAuthorityGroups ) )
-            {
-                errors.add( new ErrorReport( UserAuthorityGroup.class, ErrorCode.E3003, currentUser.getUsername(), ur.getName() ) );
-            }
-        } );
-
-        // Validate user group
-
-        boolean canAdd = currentUser.getUserCredentials().isAuthorized( UserGroup.AUTH_USER_ADD );
-
-        if ( canAdd )
-        {
-            return errors;
-        }
-
-        boolean canAddInGroup = currentUser.getUserCredentials().isAuthorized( UserGroup.AUTH_USER_ADD_IN_GROUP );
-
-        if ( !canAddInGroup )
-        {
-            errors.add( new ErrorReport( UserGroup.class, ErrorCode.E3004, currentUser ) );
-            return errors;
-        }
-
-        user.getGroups().forEach( ug ->
-        {
-            if ( ! ( currentUser.canManage( ug ) || userGroupService.canAddOrRemoveMember( ug.getUid() ) ) )
-            {
-                errors.add( new ErrorReport( UserGroup.class, ErrorCode.E3005, currentUser, ug ) );
-            }
-        } );
-
-        return errors;
-    }
-
-    @Override
-    @Transactional(readOnly = true)
-    public List<User> getExpiringUsers()
-    {
-        int daysBeforePasswordChangeRequired = (Integer) systemSettingManager.getSystemSetting( SettingKey.CREDENTIALS_EXPIRES ) * 30;
-
-        Date daysPassed = new DateTime( new Date() ).minusDays( daysBeforePasswordChangeRequired - EXPIRY_THRESHOLD ).toDate();
-
-        UserQueryParams userQueryParams = new UserQueryParams()
-            .setDisabled( false )
-            .setPasswordLastUpdated( daysPassed );
-
-        return userStore.getExpiringUsers( userQueryParams );
-    }
-
-    @Override
-    @Transactional
-    public void set2FA( User user, Boolean twoFa )
-    {
-        user.getUserCredentials().setTwoFA( twoFa );
-
-        updateUser( user );
-    }
-
-    @Override
-    public void expireActiveSessions( UserCredentials credentials )
-    {
-        List<SessionInformation> sessions = sessionRegistry.getAllSessions( credentials, false );
-
-        sessions.forEach( SessionInformation::expireNow );
-    }
-}
+package org.hisp.dhis.user;
+
+/*
+ * Copyright (c) 2004-2020, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
```

Comment:
Do you know why we get a full diff on this class?

Reply:
CRLF issue